app/models/pipeline_run_stage.rb

```diff
@@ -248,6 +248,11 @@ def db_load_host_filtering
     version_s3_path = "#{pr.host_filter_output_s3_path}/#{PipelineRun::VERSION_JSON_NAME}"
     pr.version = `aws s3 cp #{version_s3_path} -`
 
+    # Check if input was truncated
+    truncation_file = "#{pr.host_filter_output_s3_path}/#{PipelineRun::INPUT_TRUNCATED_FILE}"
+    _stdout, _stderr, status = Open3.capture3("aws", "s3", "ls", truncation_file)
+    pr.truncated = `aws s3 cp #{truncation_file} -`.split("\n")[0].to_i if status.exitstatus.zero?
```

Comment:
i'm not entirely sure about this operation - does this pull the entire file into memory to get the first line? is the file generally super small?

Reply:
Yeah, the file just contains a single number

Reply:
ah ok all good then. otherwise worth reading https://felipeelias.github.io/ruby/2017/01/02/fast-file-processing-ruby.html

Reply:
cool reference, lazy grep take 10 :)